### PR TITLE
Add doc-viewer component

### PR DESCRIPTION
**Pull request by mmalerba**

### CODE REVIEW

File: src/components/doc-viewer/doc-viewer.ts [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+        response => {
+          // TODO(mmalerba): Trust HTML.
+          let docHtml = response.text();
+          this._elementRef.nativeElement.innerHTML = docHtml;
+          this._loadLiveExamples();
+        },
+        error => {
+          console.log(error);
+          this._elementRef.nativeElement.innerText =
+          `Failed to load document: ${url}. Error: ${error}`;
```
Comment (tinayuangao) [inline src/components/doc-viewer/doc-viewer.ts]: very nit: indent

File: src/app/pages/components/components.html [hunk 2]
```diff
@@ -2,3 +2,4 @@
   <img src="../../../assets/img/components/{{component.src}}.svg" [alt]="component.name" />
   {{component.name}}
 </div>
+<doc-viewer documentUrl="/assets/documents/README.html"></doc-viewer>
```
Comment (mmalerba) [inline src/app/pages/components/components.html]: That's what I had originally, but @jelbourn suggested I change it to this

**Comment by mmalerba** [comment]
@jelbourn @tinayuangao is src/app/shared going to be the official home of our components?

**Comment by jelbourn** [comment]
@mmalerba `shared/` is fine for now, I suppose. I hadn't been paying attention to where the components lived.

**Comment by mmalerba** [comment]
tests working (though I wasn't able to tests for the ExampleViewer. I tried By.directive(ExampleViewer) but it kept returning null, I assume because of PortalHost/Portal shenanigans).

**Comment by jelbourn** [comment]
LGTM, just needs rebase

**Pull request merged**